Assets/Scripts/UI/Pickup Pup/DogBrowser.cs

```diff
@@ -0,0 +1,332 @@
+﻿/*
+ * Author(s): Isaiah Mann
+ * Description: Controls the display of a group of dogs in an organized UI
+ * Usage: Should be attached to the parent element in the UI that holds the child elements
+ */
+
+using UnityEngine;
+
+public class DogBrowser : PPUIElement 
```

Comment:
Definitely make a separate class for the button logic. We will likely need it for other screens (such as the shop: [https://drive.google.com/file/d/0BxN0RIiq3pWXNnNXaEYwbWl4SFk/]).

In general, if a class is long but not _too_ long, then regions would go a long way.